mentat/commands.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Iterable
+
+from termcolor import cprint
+
+from .errors import MentatError
+from .git_handler import commit
+
+
+class Command(ABC):
+    _registered_commands = {}
+
+    def __init_subclass__(cls, command_name: str | None) -> None:
+        if command_name is not None:
+            Command._registered_commands[command_name] = cls
+
+    @classmethod
+    def create_command(cls, command_name: str) -> Command:
+        if command_name not in cls._registered_commands:
+            return InvalidCommand(command_name)
+        else:
+            return cls._registered_commands[command_name]()
+
+    @classmethod
+    def get_command_completions(cls) -> Iterable[str]:
+        return list(map(lambda name: "/" + name, cls._registered_commands))
+
+    # *args[0] is always the command_name, c style
+    @abstractmethod
+    def apply(self, *args: str) -> None:
+        pass
+
+    # TODO: make more robust way to specify arguments for commands
```

Comment:
Wanted some outside input on this, but other than this this PR should be good to merge. I have a few ideas for how we could do arguments that I think could be better; for one, we could have an 'argumentvalidator' thing that not only would make it easier to print the argument names (i.e., it could put different brackets around required and optional arguments, as well as including defaults, etc.) but it would also let us print a usage of the command if it's misused. I also think it would be a lot better to include the command arguments in the command initialization (rather than the apply function). Let me know your thoughts on it all!

Reply:
lets have a call and maybe you can walk me through this